### ISSUE TITLE
Paste contain 0x00 characters error

### ISSUE THREAD

**Issue by kiska3** [m1dnight/exbin#42]
#### Steps to reproduce 
1. Upload text with 0x00 character eg https://transfer.archivete.am/1551Gu/31223

#### Current behaviour (bug)
Crashes when uploading text file with 0x00 character

#### Expected behaviour (correct)
Insertion into database and produce URL

#### Relevant logs and/or media (optional)
```
db_1     | 2022-01-09 15:40:58.923 AEDT [52] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
db_1     | 2022-01-09 15:40:58.923 AEDT [52] STATEMENT:  INSERT INTO "snippets" ("content","ephemeral","name","private","viewcount","inserted_at","updated_at") VALUES ($1,$2,$3,$4,$5,$6,$7) RETURNING "id"
exbin_1  | args: %{
exbin_1  |   "content" => <<83, 116, 97, 114, 116, 105, 110, 103, 32, 67, 104, 101, 99,
exbin_1  |     107, 73, 80, 32, 102, 111, 114, 32, 73, 116, 101, 109, 32, 13, 10, 70, 105,
exbin_1  |     110, 105, 115, 104, 101, 100, 32, 67, 104, 101, 99, 107, 73, 80, 32, 102,
exbin_1  |     111, 114, 32, ...>>,
exbin_1  |   "name" => "TamponArrays",
exbin_1  |   "private" => "true"
exbin_1  | }
exbin_1  | changeset: #Ecto.Changeset<
exbin_1  |   action: nil,
exbin_1  |   changes: %{
exbin_1  |     content: <<83, 116, 97, 114, 116, 105, 110, 103, 32, 67, 104, 101, 99, 107,
exbin_1  |       73, 80, 32, 102, 111, 114, 32, 73, 116, 101, 109, 32, 13, 10, 70, 105,
exbin_1  |       110, 105, 115, 104, 101, 100, 32, 67, 104, 101, 99, 107, 73, 80, 32, 102,
exbin_1  |       111, ...>>,
exbin_1  |     name: "TamponArrays"
exbin_1  |   },
exbin_1  |   errors: [],
exbin_1  |   data: #Exbin.Snippet<>,
exbin_1  |   valid?: true
exbin_1  | >
exbin_1  | 15:40:58.925 [error] Task #PID<0.4052.0> started from Exbin.Netcat terminating
exbin_1  | ** (Postgrex.Error) ERROR 22021 (character_not_in_repertoire) invalid byte sequence for encoding "UTF8": 0x00
exbin_1  |     (ecto_sql 3.6.2) lib/ecto/adapters/sql.ex:760: Ecto.Adapters.SQL.raise_sql_call_error/1
exbin_1  |     (ecto 3.6.2) lib/ecto/repo/schema.ex:725: Ecto.Repo.Schema.apply/4
exbin_1  |     (ecto 3.6.2) lib/ecto/repo/schema.ex:350: anonymous fn/15 in Ecto.Repo.Schema.do_insert/4
exbin_1  |     (ecto 3.6.2) lib/ecto/repo/schema.ex:252: Ecto.Repo.Schema.insert!/4
exbin_1  |     (ecto_sql 3.6.2) lib/ecto/adapters/sql.ex:1017: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
exbin_1  |     (db_connection 2.4.0) lib/db_connection.ex:1512: DBConnection.run_transaction/4
exbin_1  |     (exbin 0.1.6-dev.5) lib/exbin/socket.ex:64: Exbin.Netcat.serve/1
exbin_1  |     (elixir 1.12.3) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
exbin_1  | Function: #Function<0.54860568/0 in Exbin.Netcat.handle_cast/2>
exbin_1  |     Args: []

```

/label ~bug

**Comment by m1dnight** [comment]
Hi,

Thanks for submitting this. I'm currently super busy for the next month, but I promise I will have a look at this after that.

**Comment by m1dnight** [comment]
I'm currently working on this issue, and it works when I change the type to binary. This causes some problems with migrations, though, so I'm looking into how I can define this as a migration without breaking current databases. If that's not possible, I'll document on how to upgrade. 

I'll keep you posted.

**Comment by m1dnight** [comment]
Fixed it in e70ab32f3b1542dc5d20db39af2ff3589ebf8cb6.

It should work by just updating the new container. The migration will turn the column into a bytea. I also added a test for your file, and it passes. So I think this can be closed.
